src/App.tsx

```diff
@@ -1,12 +1,60 @@
 import React from 'react';
+import { Product } from './types/Product';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAllGoods, get5First, getRedGoods } from './api/goods';
+import { GoodsList } from './components/GoodsList/GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  goods: Product[];
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: [],
+  };
+
+  render() {
+    const { goods } = this.state;
+
+    return (
+      <div className="App">
+        <h1 className="App__title">Dynamic List of Goods</h1>
+
+        <button
+          type="button"
+          onClick={() => getAllGoods().then((allGoods) => {
+            this.setState({ goods: allGoods });
```

Comment:
You're repeating almost the same code in every button. You can make this a separate function 😉